File: dart/resistor-color-duo/lib/resistor_color_duo.dart
import 'dart:math';

class ResistorColorDuo {
  List<String> colors = [
    'black',
    'brown',
    'red',
    'orange',
    'yellow',
    'green',
    'blue',
    'violet',
    'grey',
    'white'
  ];
  int value(List<String> color) {
    String _result = "";
    for (int i = 0; i < min(2, color.length); i++) {
      _result += colors.indexWhere((element) => element == color[i]).toString();
    }
    return int.parse(_result);
  }
}

File: javascript/resistor-color-duo/resistor-color-duo.js
export const decodedValue = (colors) => {
  var result_string = ""
  for (let idx = 0; idx < Math.min(2, colors.length); idx++) {
    result_string = result_string + COLORS.indexOf(colors[idx]);
  }
  return Number(result_string)
};


const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];